File: app/src/main/java/com/codeartist/androidpractice/adapters/PropertyAdapter.java
package com.codeartist.androidpractice.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.codeartist.androidpractice.databinding.ItemPropertyBinding;
import com.codeartist.androidpractice.model.Property;

import java.util.List;

public class PropertyAdapter extends RecyclerView.Adapter<PropertyAdapter.ViewHolder> {

    private List<Property> properties;

    public PropertyAdapter(List<Property> employees) {
        this.properties = employees;
    }


    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemPropertyBinding binding = ItemPropertyBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(properties.get(position));

    }

    public void addProperties(Property property){
        properties.add(property);
        this.notifyItemInserted(properties.size() - 1);
    }
    public void addProperties(List<Property> properties){
        this.properties = properties;
        this.notifyDataSetChanged();
        //this.notifyItemInserted(employees.size() - 1);
    }


    @Override
    public int getItemCount() {
        return properties.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        ItemPropertyBinding binding;

        public ViewHolder(ItemPropertyBinding binding) {
            this(binding.getRoot());
            this.binding = binding;
        }

        ViewHolder(View view) {
            super(view);
        }

        void bind(@NonNull Property item) {
            binding.setProperty(item);
            binding.executePendingBindings();
        }
    }
}

File: app/src/main/java/com/codeartist/androidpractice/adapters/PropertyAdapter.kt
package com.codeartist.androidpractice.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.codeartist.androidpractice.databinding.ItemPropertyBinding
import com.codeartist.androidpractice.model.Property

class PropertyAdapter(private var properties: List<Property>) : RecyclerView.Adapter<PropertyAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = ItemPropertyBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(properties[position])
    }

   /* fun addProperties(property: Property) {
        properties.add(property)
        notifyItemInserted(properties.size - 1)
    }*/

    fun addProperties(properties: List<Property>) {
        this.properties = properties
        notifyDataSetChanged()
        //this.notifyItemInserted(employees.size() - 1);
    }

    override fun getItemCount(): Int {
        return properties.size
    }

    class ViewHolder(view: View?) : RecyclerView.ViewHolder(view!!) {
        var binding: ItemPropertyBinding? = null

        constructor(binding: ItemPropertyBinding) : this(binding.root) {
            this.binding = binding
        }

        fun bind(item: Property) {
            binding!!.property = item
            binding!!.executePendingBindings()
        }
    }
}